pep-0656.rst

```diff
@@ -0,0 +1,144 @@
+PEP: 656
+Title: Platform Tag for Linux Distributions Using Musl
+Author: Tzu-ping Chung <uranusjr@gmail.com>
+Sponsor: Brett Cannon <brett@python.org>
+PEP-Delegate: TBD
+Discussions-To: https://discuss.python.org/t/7165
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 2021-03-17
+
+
+Abstract
+========
+
+This PEP proposes a new platfrom tag series ``musllinux`` for
+binary Python package distributions for a Python installation linked
+against musl on a Linux distribution. The tag works similarly to the
+"perennial manylinux" platform tags specified in :pep:`600`, but
+targeting platforms based on musl instead.
+
+
+Motivation
+==========
+
+With the wide use of containers, distributions such as Alpine Linux,
+[alpine]_ have been gaining more popularity than ever. Many of them
+based on musl, [musl]_ a different libc implementation from glibc, and
+therefore cannot use the existing ``manylinux`` platform tags. This
+means that Python package projects cannot deploy binary distributions
+on PyPI for them. Users of such projects demand build constraints from
+those projects, putting unnecessary burden on project maintainers.
+
+
+Rationale
+=========
+
+According to the documentation, musl has a stable ABI, and maintains
+backwards compatibility, [musl-compatibility]_ [compare-libcs]_ so a
+binary compiled against an earlier version of musl is guaranteed to
+run against a newer musl runtime. [musl-compat-ml]_ Therefore, we use
+a scheme similar to the glibc-version-based manylinux tags, but
+against musl versions instead of glibc.
```

Comment:
```suggestion
According to the documentation, musl has a stable ABI, and maintains
backwards compatibility [musl-compatibility]_ [compare-libcs]_, so a
binary compiled against an earlier version of musl is guaranteed to
run against a newer musl runtime [musl-compat-ml]_. Therefore, we use
a scheme similar to the glibc-version-based manylinux tags, but
against musl versions instead of glibc.
```